addons/event/models/event_event.py

```diff
@@ -501,6 +501,10 @@ def create(self, vals_list):
         return events
 
     def write(self, vals):
+        if 'stage_id' in vals:
+            # reset kanban state when changing stage
+            if 'kanban_state' not in vals:
```

Comment:
You can probably merge the 2 conditions then:
`if 'stage_id' in vals and 'kanban_state' not in vals`